src/autopas/utils/SimilarityFunctions.h

```diff
@@ -0,0 +1,104 @@
+  }
+
+  // get cellLength which is equal in each direction, derived from the domainsize and the requested number of cells
+  double volume = domainSizePerDimension[0] * domainSizePerDimension[1] * domainSizePerDimension[2];
+  double cellVolume = volume / numberOfCells;
+  double cellLength = cbrt(cellVolume);
+
+  // calculate the size of the boundary cells, which might be smaller then the other cells
+  std::array<size_t, 3> cellsPerDimension = {};
+  // size of the last cell layer per dimension. This cell might get truncated to fit in the domain.
+  std::array<double, 3> outerCellSizePerDimension = {};
+  for (int i = 0; i < 3; ++i) {
+    outerCellSizePerDimension[i] =
+        domainSizePerDimension[i] - (floor(domainSizePerDimension[i] / cellLength) * cellLength);
+    cellsPerDimension[i] = ceil(domainSizePerDimension[i] / cellLength);
+  }
+  // Actual number of cells we end up with
+  numberOfCells = cellsPerDimension[0] * cellsPerDimension[1] * cellsPerDimension[2];
+
+  std::vector<size_t> particlesPerCell(numberOfCells, 0);
+  std::vector<double> allVolumes(numberOfCells, 0);
+
+  // add particles accordingly to their cell to get the amount of particles in each cell
+  for (auto particleItr = container->begin(autopas::IteratorBehavior::owned); particleItr.isValid(); ++particleItr) {
+    std::array<double, 3> particleLocation = particleItr->getR();
+    std::array<size_t, 3> index = {};
+    for (int i = 0; (size_t) i < particleLocation.size(); i++) {
+      index[i] = particleLocation[i] / cellLength;
+    }
+    const size_t cellIndex = autopas::utils::ThreeDimensionalMapping::threeToOneD(index, cellsPerDimension);
+    particlesPerCell[cellIndex] += 1;
+    // calculate the size of the current cell
+    allVolumes[cellIndex] = 1;
+    for (int i = 0; (size_t) i < cellsPerDimension.size(); ++i) {
+      // the last cell layer has a special size
+      if (index[i] == cellsPerDimension[i] - 1) {
+        allVolumes[cellIndex] *= outerCellSizePerDimension[i];
+      } else {
+        allVolumes[cellIndex] *= cellLength;
+      }
+    }
+  }
+
+  // calculate density for each cell
+  data[1] = 0; // max density
+  std::vector<double> densityPerCell(numberOfCells, 0.0);
+  for (int i = 0; (size_t) i < particlesPerCell.size(); i++) {
+    densityPerCell[i] =
+        (allVolumes[i] == 0) ? 0 : (particlesPerCell[i] / allVolumes[i]);  // make sure there is no division of zero
+    if (densityPerCell[i] > data[1]) {
+      data[1] = densityPerCell[i];
+    }
+  }
+
+  // get mean and reserve variable for variance
+  double mean = numberOfParticles / volume;
+  double variance = 0.0;
+
+  // calculate variance
+  for (int r = 0; (size_t) r < densityPerCell.size(); ++r) {
+    double distance = densityPerCell[r] - mean;
+    variance += (distance * distance / densityPerCell.size());
+  }
+
+  // finally calculate standard deviation
+  // return sqrt(variance);
+  data[0] = sqrt(variance);
+  return data;
```

Comment:
```suggestion
  return {homogeneighty, maxDensity};
```
The return call should then look like this.

Reply:
solved with commit "use pair instead of pointer"